### PR TITLE
Fix go dependencies

### PR DESCRIPTION
**Pull request by Racer159**
## Description

Fix our go dependency incompatibilities.

## Related Issue

Fixes #N/A

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -38,6 +38,9 @@ require (
 	sigs.k8s.io/yaml v1.3.0
 )
 
+// TODO (@WSTARR) remove this temporary replacement of oras-go 1.2.2 with defenseunicorns version due to upgraded docker lib
+replace oras.land/oras-go v1.2.2 => github.com/defenseunicorns/oras-go v1.2.3
```
Comment (jeff-mccoy) [inline go.mod:42]: nit, might be more readable if replace ops were at the top of the go.mod as we've done in the past.

File: go.mod [hunk 2]
```diff
@@ -19,16 +19,16 @@ require (
 	github.com/goccy/go-yaml v1.9.8
 	github.com/google/go-containerregistry v0.13.0
 	github.com/mholt/archiver/v3 v3.5.1
-	github.com/moby/moby v20.10.23+incompatible
+	github.com/moby/moby v23.0.1+incompatible
```
Comment (jeff-mccoy) [inline go.mod:22]: This was exploding on 23 for me before, assume this is the helm/oras replace stuff?

**Review by jeff-mccoy — COMMENTED** [review]
just a couple comments.

File: go.mod (hunk 2 above)
Comment (Racer159) [reply go.mod:22]: Correct I patched oras to work on this version of moby

File: go.mod (hunk 1 above)
Comment (Racer159) [reply go.mod]: done

**Review by jeff-mccoy — APPROVED** [review]
I see what you did there.......

**Pull request merged**